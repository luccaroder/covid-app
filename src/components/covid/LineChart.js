import React from "react";
import { ResponsiveLine } from "@nivo/line";

export default function LineChart(props){
  return (
  <div style={{height: 400}}>
    <ResponsiveLine
        data={props.chartInformation}
        margin={{ top: 50, right: 50, bottom: 50, left: 60 }}
        curve='monotoneX'
        xScale={{
          type: "time",
          format: "%Y-%m-%d %H:%M",
          precision: "hour"
        }}
        xFormat="time:%Hh"
        yScale={{ type: 'point', min: 'auto', max: 'auto' }}
        axisTop={null}
        axisRight={null}
        axisBottom={{
            orient: 'bottom',
            format: "%Hh%M",
            tickSize: 8,
            tickPadding: 5,
            tickRotation: 0,
            legendOffset: 36,
            legendPosition: 'middle',
            tickValues: 'every 5 hours'
        }}
        axisLeft={{
            orient: 'left',
            tickSize: 5,
            tickPadding: 5,
            tickRotation: 0,
            legendOffset: -40,
            legendPosition: 'middle'
        }}
        colors={{ scheme: 'category10' }}
        pointSize={10}
        pointColor='white'
        pointBorderWidth={2}
        pointBorderColor={{ from: 'serieColor' }}
        pointLabel="x"
        pointLabelYOffset={-12}
        useMesh={true}
        // legends={[
        //     {
        //         anchor: 'bottom-rigth',
        //         direction: 'column',
        //         justify: false,
        //         itemsSpacing: 0,
        //         translateX: 150,
        //         translateY: 0,
        //         itemDirection: 'left-to-right',
        //         itemWidth: 80,
        //         itemHeight: 20,
        //         itemOpacity: 0.75,
        //         symbolSize: 12,
        //         symbolShape: 'circle',
        //         symbolBorderColor: 'rgba(0, 0, 0, .5)',
        //         effects: [
        //             {
        //                 on: 'hover',
        //                 style: {
        //                     itemBackground: 'rgba(0, 0, 0, .03)',
        //                     itemOpacity: 1
        //                 }
        //             }
        //         ]
        //     }
        // ]}
    />
  </div>
)};