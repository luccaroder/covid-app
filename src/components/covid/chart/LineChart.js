import React from "react";
import { ResponsiveLine } from "@nivo/line";

export default function LineChart(props){
  return (
  <div style={{height: 400}}>
    <ResponsiveLine
        data={props.chartInformation}
        margin={{ top: 50, right: 150, bottom: 50, left: 80 }}
        yScale={{ type: "linear" }}
        xScale={{
          type: "time",
          format: "%Y-%m-%d %H:%M",
          precision: "hour"
        }}
        xFormat="time:%Hh"
        axisBottom={{
          orient: "left",
          format: "%Hh%M",
          legendOffset: 10,
          legendPosition: "middle",
          tickValues: 'every 4 hours'
        }}
        axisLeft={{
          tickValues: 8
        }}
        colors={{ scheme: 'category10' }}
        pointSize={10}
        pointColor="white"
        pointBorderWidth={2}
        pointBorderColor={{ from: 'serieColor' }}
        pointLabel="y"
        pointLabelYOffset={-12}
        useMesh={true}
        legends={[
          {
            anchor: 'bottom-right',
            direction: 'column',
            justify: false,
            translateX: 125,
            translateY: 0,
            itemsSpacing: 0,
            itemDirection: 'left-to-right',
            itemWidth: 108,
            itemHeight: 20,
            itemOpacity: 0.75,
            symbolSize: 12,
            symbolShape: 'circle',
            symbolBorderColor: 'rgba(0, 0, 0, .5)',
            effects: [
                {
                    on: 'hover',
                    style: {
                        itemBackground: 'rgba(0, 0, 0, .03)',
                        itemOpacity: 1
                    }
                }
            ]
          }
        ]}
      />
  </div>
)
    }