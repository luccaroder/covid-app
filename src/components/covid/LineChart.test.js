import React from 'react';
import LineChart from 'components/covid/LineChart'
const chartInformation = [{ x: '01:00', y: '2000' }]

jest.mock('@nivo/line', () => ({ ResponsiveLine: 'responsive-line'}))

afterEach(cleanup)

test('renders LineChart react component', () => {
    const { asFragment } = render(<LineChart chartInformation={chartInformation}/>);
    expect(asFragment(<LineChart chartInformation={chartInformation}/>)).toMatchSnapshot();
});
