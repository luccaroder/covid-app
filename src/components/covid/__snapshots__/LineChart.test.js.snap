// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LineChart react component 1`] = `
<DocumentFragment>
  <div
    style="height: 400px;"
  >
    <responsive-line
      axisbottom="[object Object]"
      axisleft="[object Object]"
      colors="[object Object]"
      curve="monotoneX"
      data="[object Object]"
      legends="[object Object]"
      margin="[object Object]"
      pointbordercolor="[object Object]"
      pointborderwidth="2"
      pointcolor="white"
      pointlabel="x"
      pointlabelyoffset="-12"
      pointsize="10"
      usemesh="true"
      xformat="time:%Hh"
      xscale="[object Object]"
      yscale="[object Object]"
    />
  </div>
</DocumentFragment>
`;
